hil-test/tests/spi_full_duplex.rs

```diff
@@ -701,9 +710,51 @@ mod tests {
         assert_eq!(&[0xff, 0xff, 0xff, 0xff], dma_rx_buf.as_slice());
     }
 
+    #[test]
+    async fn cancel_stops_basic_async_spi_transfer(mut ctx: Context) {
+        // Slow down. At 80kHz, the transfer is supposed to take a bit over 3 seconds.
+        // We don't rely on the transfer speed much, just that it's slow enough
+        // that we can detect pulses if
+        ctx.spi
+            .apply_config(&Config::default().with_frequency(Rate::from_khz(800)))
+            .unwrap();
+
+        let mut spi = ctx.spi.into_async();
+
+        for i in 0..ctx.tx_buffer.len() {
+            ctx.tx_buffer[i] = (i % 256) as u8;
+        }
+
+        let transfer = spi.transfer_in_place_async(ctx.tx_buffer);
+
+        // Wait for a bit before cancelling
+        let cancel = async {
+            for _ in 0..100 {
+                embassy_futures::yield_now().await;
+            }
+        };
```

Comment:
Ideally this would be done using PCNT to count the number of SCLK toggles, but I'm guessing we can't use PCNT for this test because the PCNT driver is unstable?

Reply:
It's not PCNT because not all devices have PCNT. It's not an SPI slave either, for the same reason. We only need a yes/no decision here, no counting, no data reception.

Stability is the excuse for the yield loop only.

Reply:
Perhaps if the transferred data looked like this `[0, 0, 0, 0, 0, 0, 0, 0, 0, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF]`.
You can then do something like `miso_loopback.wait_for_high()` rather than the for loop of yields.

The magic number of 100 is making want to find alternatives 😅, as I can't tell what you're waiting for.

Reply:
Apparently. That loop of yields is an alternative for `embassy_time::Timer::after().await`. I expect no edges to happen for _some time_, so the transferred data doesn't come into play here.

Reply:
Ah right, got it